src/components/content-tab-snapshots.tsx

```diff
@@ -390,14 +396,28 @@ function AddDemoSiteWithProgress( {
 		interval: 1500,
 		maxValue: 95,
 	} );
+
+	useEffect( () => {
+		const intervalId = setInterval( () => {
+			fetchSnapshotUsage(); // Function to fetch the latest data from the endpoint
+		}, 10000 );
```

Comment:
The only way I found to get the real-time data was to poll the endpoint - not sure if it is the best solution. Perhaps we can reduce the polling interval as the case of the user being blocked does not seem to be something that would happen often

Reply:
Here's a potential approach:

```patch
diff --git forkSrcPrefix/src/hooks/use-snapshots.tsx forkDstPrefix/src/hooks/use-snapshots.tsx
index 564090ea621e4144ab68922c4083ffb66cb17cb8..32abc542d3d40f057290ac7c044c6c1ffc270587 100644
--- forkSrcPrefix/src/hooks/use-snapshots.tsx
+++ forkDstPrefix/src/hooks/use-snapshots.tsx
@@ -33,7 +33,9 @@ interface SnapshotContextType {
 	loadingServerSnapshots: boolean;
 	activeSnapshotCount: number;
 	snapshotQuota: number;
-	fetchSnapshotUsage: () => Promise< { site_limit: number; site_count: number } | undefined >;
+	fetchSnapshotUsage: () => Promise<
+		{ site_limit: number; site_count: number; site_creation_blocked: boolean } | undefined
+	>;
 	isLoadingSnapshotUsage: boolean;
 	initiated: boolean;
 	snapshotCreationBlocked: boolean;
diff --git forkSrcPrefix/src/components/content-tab-snapshots.tsx forkDstPrefix/src/components/content-tab-snapshots.tsx
index 8a1d8c3571b78a26e85868f0b1dc57fe1fa61be8..3ffa66b760599f11da7d755f82bb40c925d0b93a 100644
--- forkSrcPrefix/src/components/content-tab-snapshots.tsx
+++ forkDstPrefix/src/components/content-tab-snapshots.tsx
@@ -381,13 +381,8 @@ function AddDemoSiteWithProgress( {
 	const { __, _n } = useI18n();
 	const { archiveSite, isUploadingSiteId, isAnySiteArchiving } = useArchiveSite();
 	const isUploading = isUploadingSiteId( selectedSite.id );
-	const {
-		activeSnapshotCount,
-		snapshotQuota,
-		isLoadingSnapshotUsage,
-		snapshotCreationBlocked,
-		fetchSnapshotUsage,
-	} = useSnapshots();
+	const { activeSnapshotCount, snapshotQuota, isLoadingSnapshotUsage, snapshotCreationBlocked } =
+		useSnapshots();
 	const isLimitUsed = activeSnapshotCount >= snapshotQuota;
 	const isOffline = useOffline();
 	const { progress, setProgress } = useProgressTimer( {
@@ -397,14 +392,6 @@ function AddDemoSiteWithProgress( {
 		maxValue: 95,
 	} );
 
-	useEffect( () => {
-		const intervalId = setInterval( () => {
-			fetchSnapshotUsage(); // Function to fetch the latest data from the endpoint
-		}, 10000 );
-
-		return () => clearInterval( intervalId ); // Cleanup the interval on component unmount
-	}, [ fetchSnapshotUsage ] );
-
 	useEffect( () => {
 		if ( isSnapshotLoading ) {
 			setProgress( 80 );
diff --git forkSrcPrefix/src/hooks/use-archive-site.ts forkDstPrefix/src/hooks/use-archive-site.ts
index 02061b044032c14fc4fcfc001828eaae465107c1..6d0d5795cc0073d6121f0feb59e8d2db8e4f97b5 100644
--- forkSrcPrefix/src/hooks/use-archive-site.ts
+++ forkDstPrefix/src/hooks/use-archive-site.ts
@@ -12,7 +12,7 @@ import { SnapshotStatus, SnapshotStatusResponse, useSnapshots } from './use-snap
 
 export function useArchiveSite() {
 	const { uploadingSites, setUploadingSites } = useSiteDetails();
-	const { snapshots, addSnapshot, updateSnapshot } = useSnapshots();
+	const { snapshots, addSnapshot, updateSnapshot, fetchSnapshotUsage } = useSnapshots();
 	const isUploadingSiteId = useCallback(
 		( localSiteId: string ) => uploadingSites[ localSiteId ] || false,
 		[ uploadingSites ]
@@ -70,6 +70,14 @@ export function useArchiveSite() {
 			}
 
 			setUploadingSites( ( _uploadingSites ) => ( { ..._uploadingSites, [ siteId ]: true } ) );
+
+			const snapshotUsage = await fetchSnapshotUsage();
+			if ( snapshotUsage?.site_creation_blocked ) {
+				alert( errorMessages.rest_site_creation_blocked );
+				setUploadingSites( ( _uploadingSites ) => ( { ..._uploadingSites, [ siteId ]: false } ) );
+				return;
+			}
+
 			const { zipContent, zipPath, exceedsSizeLimit } = await getIpcApi().archiveSite( siteId );
 			if ( exceedsSizeLimit ) {
 				alert(

```

Reply:
The main downside of the above approach compared to polling the API would be for the case of blocked users being unblocked. As the demo site creation won't be enabled until they restart the app. In any case, I presume this will be an uncommon case that we could omit in the implementation. Alternatively, we could check usage when the app gains focus.